### PR TITLE
[HALX86] Convert the PCI Ids database files to LF ('\n') endlines, and adjust the parsing code.

### PR DESCRIPTION
**Pull request by HBelusca**
## Purpose

This allows to somewhat reduce their size, and the resulting size of the generated hal.dll file.
Cc @disean too (you're not in the reviewers list :thinking: )

pci_classes.ids
Before: 6,207 bytes
After : 5,968 bytes
Reduction: 239 bytes --> 4%

pci_vendors.ids
Before: 1,370,248 bytes
After : 1,334,622 bytes
Reduction: 35,626 bytes --> 2.6%

### CODE REVIEW

File: hal/halx86/legacy/bussupp.c
Comment (HBelusca) [inline hal/halx86/legacy/bussupp.c:833]: Note: In the code here and above I'm not fully satisfied with the single alone '\n' ; I would like it to be function of `NEWLINE` instead; any help about what needs to be slightly adjusted (I suppose, with +-1 offsets or `>` -> `>=` types of changes) would be welcome.

Comment (binarymaster) [reply hal/halx86/legacy/bussupp.c:833]: Doesn't `strlen(NEWLINE "\t00  ");` just work here? See also lines 820-821.

Comment (binarymaster) [reply hal/halx86/legacy/bussupp.c:833]: If I remember correctly, a single `\n` was used here just for compactness.

File: hal/halx86/legacy/bussupp.c
Comment (JoachimHenze) [inline hal/halx86/legacy/bussupp.c:907]: Couldn't we write:
Length = min(p - SubVendorName, sizeof(bSubVendorName) - 1);
at line 907 and entirely omit line 906.

Comment (HBelusca) [reply hal/halx86/legacy/bussupp.c:907]: Yes we could, but since `min` (and `max`) is a macro: `#define min(a, b) ((a) < (b) ? (a) : (b))` for instance, it can evaluate more than once its arguments, so usually I avoid doing calculations within the min/max arguments (unless their result is known at compile-time, like the sizeof(xxx)-1).
(Also this makes the code line a bit shorter...)

File: hal/halx86/legacy/bussupp.c
Comment (binarymaster) [reply hal/halx86/legacy/bussupp.c]: I've replaced remaining NEWLINE's in the second commit, please retest. I think there should be no problem.

File: hal/halx86/legacy/bussupp.c
Comment (JoachimHenze) [reply hal/halx86/legacy/bussupp.c:907]: Your reasoning for trying to avoid the macros here does seem very legit to me. Thank you for explaining. Resolved.

**Review by JoachimHenze — APPROVED** [review]
Approved. Although I want to emphasize that @binarymaster has pushed another change on top here today, where i am not entirely sure, whether it is safe.
In general I think it is extremely unpolite to push anything into other peoples PRs without explicit consent beforehand. Better use the suggestion-feature or a comment and let the PRs author decide himself whether he wants to accept such a change.

So unless this was discussed with @HBelusca beforehand, I do heavily recommend @HBelusca to review those last-second-changes again before merging this PR.

**Review by binarymaster — COMMENTED** [review]
I'm thinking about adding automated tests for this function with a set of predefined known good values, just to be sure nothing has changed.

**Comment by JoachimHenze** [comment]
@binarymaster Writing a test is a very good idea. Still I think such pushes without prior authors approval are generally something to avoid at all costs if you don't want to piss off the contributor. In some cases people might have different preferences.
Exception of such a moral rule are orphaned PRs ofc where the contributor cannot be contacted anymore.

**Comment by HBelusca** [comment]
(PR rebased on top of current master).

**Comment by HBelusca** [comment]
PR is not perfect yet:

- Before:
```
00:00.0 Host bridge [0600]: Intel Corporation 440FX - 82441FX PMC [Natoma] [8086:1237] (rev 02)
        Subsystem: Unknown [0000:0000]
        Flags: fast devsel, latency 0
        Device has Extended Address Decoding. It may fail to work on older BIOSes!

[...]
```

- After (with or without @binarymaster 's addition):
```
 [8086:1237] (rev 02)[Natoma]
        Subsystem: Unknown [0000:0000]
        Flags: fast devsel, latency 0
        Device has Extended Address Decoding. It may fail to work on older BIOSes!

[...]
```

I think that the extra '\r' characters now pass through what's being displayed, thus messing the display.

**Comment by HBelusca** [comment]
OK after retesting (and ensuring the .IDS files have their correct newlines), I get, for my VM, the same result as without the PR modifications applied:
```
====== PCI BUS HARDWARE DETECTION =======

00:00.0 Host bridge [0600]: Intel Corporation 440FX - 82441FX PMC [Natoma] [8086:1237] (rev 02)
        Subsystem: Unknown [0000:0000]
        Flags: fast devsel, latency 0
        Device has Extended Address Decoding. It may fail to work on older BIOSes!

00:01.0 ISA bridge [0601]: Intel Corporation 82371SB PIIX3 ISA [Natoma/Triton II] [8086:7000] (rev 00)
        Subsystem: Unknown [0000:0000]
        Flags: bus master, medium devsel, latency 0

00:01.1 IDE interface [0101]: Intel Corporation 82371AB/EB/MB PIIX4 IDE [8086:7111] (rev 01)
        Subsystem: Unknown [0000:0000]
        Flags: bus master, fast devsel, latency 0
        I/O ports at d000 [size=16]

00:02.0 VGA compatible controller [0300]: InnoTek Systemberatung GmbH VirtualBox Graphics Adapter [80ee:beef] (rev 00)
        Subsystem: Unknown [0000:0000]
        Flags: bus master, fast devsel, latency 0, IRQ 11
        Memory at e0000000 (32-bit, prefetchable) [size=16M]
        Device is using IRQ 11! ISA Cards using that IRQ may fail!

00:03.0 Ethernet controller [0200]: Advanced Micro Devices, Inc. [AMD] 79c970 [PCnet32 LANCE] [1022:2000] (rev 40)
        Subsystem: PCnet - Fast 79C971 [1022:2000]
        Flags: bus master, medium devsel, latency 0, IRQ 10
        I/O ports at d020 [size=32]
        Memory at f0000000 (32-bit, non-prefetchable) [size=4K]
        Device is using IRQ 10! ISA Cards using that IRQ may fail!

00:04.0 System peripheral [0880]: InnoTek Systemberatung GmbH VirtualBox Guest Service [80ee:cafe] (rev 00)
        Subsystem: Unknown [0000:0000]
        Flags: bus master, fast devsel, latency 0, IRQ 09
        I/O ports at d040 [size=32]
        Memory at f0400000 (32-bit, non-prefetchable) [size=4M]
        Memory at f0800000 (32-bit, prefetchable) [size=16K]
        Device is using IRQ 9! ISA Cards using that IRQ may fail!

00:05.0 Multimedia audio controller [0401]: Intel Corporation 82801AA AC'97 Audio Controller [8086:2415] (rev 01)
        Subsystem: Unknown [8086:0000]
        Flags: bus master, medium devsel, latency 0, IRQ 11
        I/O ports at d100 [size=256]
        I/O ports at d200 [size=64]
        Device is using IRQ 11! ISA Cards using that IRQ may fail!

00:06.0 USB controller [0c03]: Apple Inc. KeyLargo/Intrepid USB [106b:003f] (rev 00)
        Subsystem: Unknown [0000:0000]
        Flags: bus master, fast devsel, latency 0, IRQ 11
        Memory at f0804000 (32-bit, non-prefetchable) [size=4K]
        Device is using IRQ 11! ISA Cards using that IRQ may fail!

00:07.0 Bridge [0680]: Intel Corporation 82371AB/EB/MB PIIX4 ACPI [8086:7113] (rev 08)
        Subsystem: Unknown [0000:0000]
        Flags: bus master, medium devsel, latency 0, IRQ 09
        Device is using IRQ 9! ISA Cards using that IRQ may fail!

00:0b.0 USB controller [0c03]: Intel Corporation 82801FB/FBM/FR/FW/FRW (ICH6 Family) USB2 EHCI Controller [8086:265c] (rev 00)
        Subsystem: Unknown [0000:0000]
        Flags: bus master, fast devsel, latency 0, IRQ 10
        Memory at f0805000 (32-bit, non-prefetchable) [size=4K]
        Device is using IRQ 10! ISA Cards using that IRQ may fail!

====== PCI BUS DETECTION COMPLETE =======
```

**Comment by HBelusca** [comment]
> I'm thinking about adding automated tests for this function with a set of predefined known good values, just to be sure nothing has changed.

Do you want to do that in a separate PR @binarymaster ?

**Comment by binarymaster** [comment]
> Do you want to do that in a separate PR?

Well I wanted to just test that locally without the intention of creating an API test, but as I can see you have already tested it. :smile:

**Pull request merged**
